Sources/PlatformViewVersion.swift

```diff
@@ -1,31 +1,44 @@
 import SwiftUI
 
-public struct PlatformViewVersions<SwiftUIViewType: IntrospectableViewType, PlatformSpecificEntity: PlatformEntity> {
-    let isCurrent: Bool
+public struct PlatformViewVersionPredicate<SwiftUIViewType: IntrospectableViewType, PlatformSpecificEntity: PlatformEntity> {
     let selector: IntrospectionSelector<PlatformSpecificEntity>?
 
     private init<Version: PlatformVersion>(
-        _ versions: [PlatformViewVersion<Version, SwiftUIViewType, PlatformSpecificEntity>]
+        _ versions: [PlatformViewVersion<Version, SwiftUIViewType, PlatformSpecificEntity>],
+        matches: (PlatformViewVersion<Version, SwiftUIViewType, PlatformSpecificEntity>) -> Bool
     ) {
-        if let currentVersion = versions.first(where: \.isCurrent) {
-            self.isCurrent = true
-            self.selector = currentVersion.selector
+        if let matchingVersion = versions.first(where: matches) {
+            self.selector = matchingVersion.selector ?? .default
         } else {
-            self.isCurrent = false
             self.selector = nil
         }
     }
 
     public static func iOS(_ versions: (iOSViewVersion<SwiftUIViewType, PlatformSpecificEntity>)...) -> Self {
-        Self(versions)
+        Self(versions, matches: \.isCurrent)
+    }
+
+    @_spi(Advanced)
```

Comment:
I guess the question is... should we still gate these? Maybe not 🤔

Reply:
I decided to keep it, and also to rename existing `@_spi(Internals)` annotations to `@_spi(Advanced)`, so everything is accessible from one entry point. Also "Advanced" is a better name than "Internals" because it denotes intent ("Advanced Mode").

Reply:
Sry for the late reply! I definitely think `Advanced` is such a better name. Internal really gives me "not intended for use" vibes.
With that convention I would say it encourages more people to use. 
(`Internal` also made me a bit unsure and hence the question in https://github.com/siteline/SwiftUI-Introspect/discussions/280)